snuba/consumers/consumer.py

```diff
@@ -601,15 +605,75 @@ class MultistorageKafkaPayload(NamedTuple):
 ]
 
 
+def __message_to_dict(
+    message: Message[KafkaPayload],
+) -> Dict[str, Any]:
+    # TODO: Remove temporary mitigations for DLQ prod test
+    return {
+        "message_payload_value": message.payload.value,
+        "message_offset": message.offset,
+        "message_partition": message.partition.index,
+        "message_topic": message.partition.topic.name,
+    }
+
+
+def skip_kafka_message(message: Message[KafkaPayload]) -> bool:
+    # TODO: Remove temporary mitigations for DLQ prod test
+
+    # expected format is "[topic:partition_index:offset,...]" eg [snuba-metrics:0:1,snuba-metrics:0:3]
+    messages_to_skip = (get_config("kafka_messages_to_skip") or "[]")[1:-1].split(",")
+    return (
+        f"{message.partition.topic.name}:{message.partition.index}:{message.offset}"
+        in messages_to_skip
+    )
+
+
 def process_message(
     processor: MessageProcessor, message: Message[KafkaPayload]
 ) -> Union[None, BytesInsertBatch, ReplacementBatch]:
-    result = processor.process_message(
-        rapidjson.loads(message.payload.value),
-        KafkaMessageMetadata(
-            message.offset, message.partition.index, message.timestamp
-        ),
-    )
+
+    # TODO: Remove temporary mitigations for DLQ prod test
+
+    # Mitigation: Last resort - Hardcode message(s) to skip in runtime config
+    if skip_kafka_message(message):
+        logger.warning(
+            f"A consumer for {message.partition.topic.name} skipped a message!",
+            exc_info=True,
+            extra=__message_to_dict(message),
+        )
+        return None
+
+    # Mitigation: JSON fails to deserialize
+    try:
+        deserialized_message = rapidjson.loads(message.payload.value)
+    except Exception as err:
+        logger.error(
+            err,
+            exc_info=True,
+            extra=__message_to_dict(message),
+        )
+        return None
+
+    # Mitigation: An unexpected exception is raised by a Metrics processor
+    try:
+        result = processor.process_message(
+            deserialized_message,
+            KafkaMessageMetadata(
+                message.offset, message.partition.index, message.timestamp
+            ),
+        )
+    except Exception as err:
+        if (
+            message.partition.topic.name == StreamsTopic.METRICS.value
+            and not isinstance(err, InvalidMessages)
```

Comment:
I don't think this really solves the problem. 
While you would not make the metrics consumer crash, you would not send errors to the DLQ that are not of the 2-3 specific types that the metrics consumer turn into InvalidMessage.

It seems to me that this could be much simpler by just catching the exception coming from process_message, turning it into an `InvalidMessage` and raising it again. Also remove the specific logic from the metrics processor.